sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs

```diff
@@ -39,14 +39,22 @@ internal FormField(string name, FieldValue_internal fieldValue, IReadOnlyList<Re
             Name = name;
             LabelData = null;
 
-            IReadOnlyList<FormElement> FormElement = fieldValue.Elements != null
-                ? ConvertTextReferences(fieldValue.Elements, readResults)
-                : new List<FormElement>();
+            // Bounding box, page and text are not returned by the service in two scenarios:
+            //   - When this field is global and not associated with a specific page (e.g. ReceiptType).
+            //   - When this field is a collection, such as a list or dictionary.
+            //
+            // In these scenarios we do not set a ValueData.
+
+            if (fieldValue.BoundingBox.Count != 0 || fieldValue.Page != null || fieldValue.Text != null)
+            {
+                IReadOnlyList<FormElement> FormElement = ConvertTextReferences(fieldValue.Elements, readResults);
 
-            // TODO: FormEnum<T> ?
-            BoundingBox boundingBox = fieldValue.BoundingBox == null ? default : new BoundingBox(fieldValue.BoundingBox);
+                // TODO: FormEnum<T> ?
+                BoundingBox boundingBox = new BoundingBox(fieldValue.BoundingBox);
+
+                ValueData = new FieldData(fieldValue.Text, fieldValue.Page.Value, boundingBox, FormElement);
+            }
```

Comment:
`fieldValue.Elements` and `fieldValue.BoundingBox` are non-nullable, according to the swagger file, so we're skipping null checks here. When these values are not returned by the service, CodeGen already creates them as empty lists for us.

Reply:
>fieldValue.Elements and fieldValue.BoundingBox are non-nullable

>fieldValue.Elements and fieldValue.BoundingBox are non-nullable

I'm confused... Is this then a mistake in the swagger?
Will it be better to make a transformation in the swagger and modify this? I worry that if they change the swagger, codegen changes, and then we will might forget to add the extra check

Reply:
It took me more time than I wanted it to to understand how CodeGen deals with the response returned by the service. In my understanding, there are three possible scenarios when receiving a collection from the service:

1. **The collection is returned and it's not null:**

    - Always allowed.
    - CodeGen gives us the actual bounding box returned by the service. No surprises here.

2. **The collection is returned and it's null:**

    - Not allowed by default. Property must explicitly set `x-nullable = true` in the swagger to enable this scenario.
    - CodeGen gives us a null reference.
    - Not supported by `fieldValue.boundingBox`. The service works exactly as described in the swagger.

3. **The collection is not returned:**

    - Allowed by default. Property must be explicitly marked as `required` in the swagger to disable this scenario.
    - CodeGen gives us an empty collection.
    - Supported by `fieldValue.boundingBox`. That's why instead of a null check we're verifying the count of elements.

To directly answer your question: `fieldValue.elements` and `fieldValue.boundingBox` are non-nullable, but this doesn't mean they must be returned. When they're not, we'll have empty lists instead of null references.

Reply:
> I worry that if they change the swagger, codegen changes, and then we will might forget to add the extra check

I feel your pain, specially because the service underwent multiple changes recently. However, if the service starts behaving differently, I would prefer throwing, understanding what has changed and implementing a fix with the expected approach. The service may intentionally want to differentiate between "null" and "no value returned".

I agree that in most cases we'll probably just add a null check and keep doing the same thing, but I think that throwing is better than letting a silent bug to be introduced.

Reply:
Fair. We will need to understand the implications of the service making the change. 
Thanks for the detailed explanation!!!